File: mirea/contests/2021-09-26 MIREA Welcome Contest 2021/NoPoint C++/Задача D.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  const int LETTERS = 26;
  int tests;
  cin >> tests;
  for(int test = 0; test < tests; ++test) {
    vector<string> ciphered(LETTERS);
    for(int i = 0; i < LETTERS; ++i) {
      cin >> ciphered[i];
    }
    string s1, s2;
    cin >> s1 >> s2;
    int m = (int)s2.size();

    string res;
    for(int i = 0; i < m; ++i) {
      res.append(ciphered[s2[i] - 'a']);
    }
    bool ans = (res == s1);
    cout << (ans ? "Yes" : "No") << endl;
  }
}

File: mirea/contests/2021-09-26 MIREA Welcome Contest 2021/NoPoint Python/Задача D.py
def main():
  ALPH = "abcdefghijklmnopqrstuvwxyz"
  LETTERS = len(ALPH)
  TO_NUM = dict(zip(ALPH, range(LETTERS)))

  tests = int(input())
  for test in range(tests):
    chars = [input() for _ in range(LETTERS)]
    ciphered = input()
    message = input()
    
    my_ciphered = ''.join((chars[TO_NUM[char]] for char in message))
    print("Yes" if ciphered == my_ciphered else "No")


main()

